addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -327,102 +449,51 @@ odoo.define('website_slides.fullscreen', function (require) {
-            self.goToQuiz = true;
-            self._setPreviousAndNextSlides();
-            self._renderPlayer();
-            self._setActiveTab();
-            self._updateUrl();
-            history.pushState(null,'' ,self.url);
-        },
-         /**
-        * @private
-        * Binds events related to the list
-        */
-        _bindListEvents: function(){
-            var self = this;
-            $('.o_wslides_fullscreen_sidebar_slide_tab').each(function () {
-                $(this).click(self._onListCellClick.bind(self));
-            });
-
-            $('.o_wslides_slide_quiz ').each(function(){
-                $(this).click(self._onMiniQuizClick.bind(self));
-            })
+        _onClickSlideTab: function (ev){
+            this.goToQuiz = false;
+            this._changeSlide(ev);
         },
-        _onKeyDown: function(ev){
-            var self = this;
-            switch(ev.key){
-                case "ArrowRight":
-                self._goToNextSlide();
-                break;
-                case "ArrowLeft":
-                self._goToPreviousSlide();
-                break;
-            }
+        _onToggleSidebar: function (){
+            this.sidebar.toggle();
+            this.$('.o_wslides_fs_player').toggleClass('o_wslides_fs_player_no_sidebar');
         },
-    })
+    });
 
     sAnimations.registry.websiteSlidesFullscreenPlayer = Widget.extend({
-        selector: '.oe_js_course_slide',
+        selector: '.o_wslides_fullscreen',
         xmlDependencies: ['/website_slides/static/src/xml/website_slides_fullscreen.xml'],
-        init: function(el){
+        init: function (el){
             this._super.apply(this, arguments);
         },
-        start: function(){
-            this._super.apply(this, arguments);
-            var user_id = this.$el.attr('user_id');
-            var course_id = this.$el.attr('course_id');
-            var slide_id = this.$el.attr('slide_id');
-            var fullscreen = new Fullscreen(this, course_id, slide_id, user_id);
-            fullscreen.appendTo(".oe_js_course_slide");
-        }
+        start: function (){
+            var defs = [this._super.apply(this, arguments)];
+            var fullscreen = new Fullscreen(this, this._getSlides(), this._getCurrentSlideIndex());
+            defs.push(fullscreen.attachTo(".o_wslides_fullscreen"));
+            return $.when.apply($, defs);
+        },
+        _getCurrentSlideIndex: function (){
+            return parseInt(this.$('.o_wslides_fs_sidebar_slide_tab.active').attr('index'));
+        },
+        /**
+        * @private
+        * Creates slides objects from every slide-list-cells attributes
+        */
+       _getSlides: function (){
+            var slides = $('.o_wslides_fs_sidebar_slide_tab');
+            var slideList = [];
+            for (var i = 0; i < slides.length;i++){
+                var slide = $(slides[i]);
+                slideList.push(slide.data());
```

Comment:
As data here, we should have name, type, hasQuiz, completed, id, ... and some specific (embed url, ...) for specific type.